bson/vector.go

```diff
@@ -266,3 +275,60 @@ func newBitVector(b []byte) (Vector, error) {
 	}
 	return NewPackedBitVector(b[1:], b[0])
 }
+
+// DecodeVectorInt8 decodes a BSON Vector binary value (subtype 9) into a []int8 slice.
+// The binary data should be in the format: [<vector type> <padding> <data>]
+// For int8 vectors, the vector type is 0x01.
+func DecodeVectorInt8(data []byte) ([]int8, error) {
```

Comment:
`DecodeVectorInt8` and `DecodeVectorFloat32` should not be exported and they should be migrated to slice_codec.go.